geomesa-spark/geomesa-spark-sql/src/main/scala/org/apache/spark/sql/SQLRules.scala

```diff
@@ -71,6 +73,24 @@ object SQLRules extends LazyLogging {
     }
   }
 
+  def sedonaExprToGTFilter(pred: Sedona_ST_Predicate): Option[GTFilter] = {
+    val left = pred.children.head
+    val right = pred.children.last
+    (sparkExprToGTExpr(left), sparkExprToGTExpr(right)) match {
+      case (Some(expr1), Some(expr2)) => pred match {
+        case Sedona_ST_Contains(_) => Some(ff.contains(expr1, expr2))
+        case Sedona_ST_Crosses(_) => Some(ff.crosses(expr1, expr2))
+        case Sedona_ST_Overlaps(_) => Some(ff.overlaps(expr1, expr2))
+        case Sedona_ST_Intersects(_) => Some(ff.intersects(expr1, expr2))
```

Comment:
After running the code:
```bash

git clone git@github.com:Kontinuation/geomesa.git
cd geomesa
git checkout sedona-integration-pr

build/mvn clean install -T8 -DskipTests
```

I fail to get the functions registered:
```
spark-shell --packages "org.locationtech.geomesa:geomesa-spark-sql_2.12:3.4.0-SNAPSHOT,org.apache.sedona:sedona-python-adapter-3.0_2.12:1.1.1-incubating" --conf "spark.serializer=org.apache.spark.serializer.KryoSerializer" --conf "spark.kryo.registrator=import org.locationtech.geomesa.spark.GeoMesaSparkKryoRegistrator" --conf "spark.geomesa.sedona.udf.prefix=sedona_"

Class.forName("org.apache.sedona.core.serde.SedonaKryoRegistrator")
Class.forName("org.apache.spark.sql.sedona_sql.UDT.GeometryUDT")

import org.locationtech.geomesa.spark.jts
jts.initJTS(spark.sqlContext)

spark.sessionState.functionRegistry.listFunction.map(_.funcName).sorted.filter(_.toUpperCase.contains("ST_")).foreach(println)
// all the geomesa stuff is here
spark.sessionState.functionRegistry.listFunction.map(_.funcName).sorted.filter(_.toUpperCase.contains("SEDONA")).foreach(println)
// no SEDONA things are here
```

even though:

`mvn -pl org.locationtech.geomesa:geomesa-spark-test-with-sedona_2.12 clean package`

succeeds (and thus can load) the sedona functions.

Reply:
you might need one of the runtime jars, e.g. geomesa-gt-spark-runtime, those are uber shaded jars with all the required dependencies.

Reply:
https://github.com/locationtech/geomesa/pull/2802/files#diff-8b5907286a459c13b7639108a67cb0aadb6846f1fa79df21cb21cb72173c26b2R91 is looking useful

Reply:
SQLTypes.init(spark.sqlContext)

Reply:
will --packages pull from non-standard repositories? geotools is in it's own maven repo.

Reply:
no  - you would need to specify the resolvers. However, as I want to use geomesa`s snapshot for the PR I needed to install it to the local M2 repository first. Then everything is pulled already by maven and spark directly resolves locally

Reply:
> SQLTypes.init(spark.sqlContext)

Yes. Sedona UDFs and UDAFs were registered in [`SQLTypes.init`](https://github.com/locationtech/geomesa/pull/2802/files#diff-9cf5352519b4798ab6e5e89f812d98784c45138e99efe3ff67350fb75eb8cd7eR17-R23).

I need to emphasize this in the documentation by adding extra explanations to the [example code for creating Spark session object](https://github.com/locationtech/geomesa/pull/2802/files#diff-b165c48f8f912cd9b059e05b8150825e34c3dfe9e38a7059e202efeaa8728de1R307-R327).